packages/beacon-node/src/network/events.ts

```diff
@@ -27,13 +28,13 @@ export enum NetworkEvent {
 }
 
 export type NetworkEventData = {
-  [NetworkEvent.peerConnected]: {peer: PeerIdStr; status: phase0.Status};
-  [NetworkEvent.peerDisconnected]: {peer: PeerIdStr};
-  [NetworkEvent.reqRespRequest]: {request: RequestTypedContainer; peer: PeerId};
-  [NetworkEvent.unknownBlockParent]: {blockInput: BlockInput; peer: PeerIdStr};
-  [NetworkEvent.unknownBlock]: {rootHex: RootHex; peer?: PeerIdStr};
-  [NetworkEvent.pendingGossipsubMessage]: PendingGossipsubMessage;
-  [NetworkEvent.gossipMessageValidationResult]: {
+  [NetworkEvent.peerConnected]: EmittedAt & {peer: PeerIdStr; status: phase0.Status};
+  [NetworkEvent.peerDisconnected]: EmittedAt & {peer: PeerIdStr};
+  [NetworkEvent.reqRespRequest]: EmittedAt & {request: RequestTypedContainer; peer: PeerId};
+  [NetworkEvent.unknownBlockParent]: EmittedAt & {blockInput: BlockInput; peer: PeerIdStr};
+  [NetworkEvent.unknownBlock]: EmittedAt & {rootHex: RootHex; peer?: PeerIdStr};
+  [NetworkEvent.pendingGossipsubMessage]: EmittedAt & PendingGossipsubMessage;
+  [NetworkEvent.gossipMessageValidationResult]: EmittedAt & {
```

Comment:
Instead of adding this property on the business logic you can just track timings on the event bridge wrapper object

Reply:
Updated.